### PR TITLE
Add support to speed limit penalty in Carla

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
This PR is only on adding overspeed reward/penalty, without modifying observations.
Observation part can be done separately if required.

### CODE REVIEW

File: alf/environments/suite_carla.py [hunk 1]
```diff
@@ -745,6 +792,14 @@ def get_current_time_step(self, current_frame):
                 Player.PENALTY_RATE_RED_LIGHT * max(0., self._episode_reward))
         self._prev_violated_red_light_id = red_light_id
 
+        overspeed = self.get_overspeed_amount()
+        if overspeed > 0:
+            logging.info("actor=%d frame=%d OVERSPEED" % (self._actor.id,
+                                                          current_frame))
+            reward_vector[Player.REWARD_OVERSPEED] = 1.
+            info['overspeed'] = np.float32(1.0)
+            reward -= self._overspeed_penalty_weight * overspeed
```
Comment (emailweixu) [inline alf/environments/suite_carla.py]: To make this invariant to delta_seconds, it needs to be multiplied with ``self._delta_seconds`` thus with same unit as distance.

File: alf/environments/suite_carla.py [hunk 2]
```diff
@@ -615,6 +626,41 @@ def _get_goal(self):
         return _calculate_relative_position(self._actor.get_transform(),
                                             self._goal_location)
 
+    def update_speed_limit(self, dis_threshold=1.0):
+        """Update the speed limit of the actor according to the active speed
+        limit sign. The speed limit is updated when passing by a speed limit sign.
+
+        Args:
+            dis_threshold (float): the distance within which to consider the
```
Comment (emailweixu) [inline alf/environments/suite_carla.py]: It the unit meter? I feel 1 meter is too low. It may miss the speed limit sign.

**Review by emailweixu — CHANGES_REQUESTED** [review]

File: alf/environments/suite_carla.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_carla.py]: Good point. Done.

File: alf/environments/suite_carla.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/environments/suite_carla.py]: The unit is meter. Doc str has been updated. The threshold has been increased to a larger value.

**Review by emailweixu — APPROVED** [review]

**Comment by Haichao-Zhang** [comment]
For reference, here is a list of all speed limit signs and their locations in Town01. The unit of speed limit is km/h:
image

**Pull request merged**